File: test_dynamic_agents.py
#!/usr/bin/env python3
"""
Test script to verify agents are using database settings instead of static files
"""
import sys
import os
sys.path.append('/app')

from src.utils.dynamic_agent_loader import dynamic_loader
from src.distributed_router import DistributedRouter
from src.config import config

def test_dynamic_agents():
    print("🧪 Testing Dynamic Agent Loading...")
    
    # Initialize router
    router = DistributedRouter()
    
    # Get available roles from database
    roles = dynamic_loader.get_available_roles()
    print(f"📋 Available agent roles in database: {roles}")
    
    if not roles:
        print("❌ No agents found in database. Run import first!")
        return
    
    # Test creating an agent from database
    test_role = roles[0] if roles else "Project Manager"
    print(f"\n🔧 Testing agent creation for role: {test_role}")
    
    # Get config from database
    config_data = dynamic_loader.get_agent_config(role=test_role)
    if config_data:
        print(f"✅ Found database config for {test_role}:")
        print(f"   - Memory: {config_data.get('memory', 'Not set')}")
        print(f"   - Learning: {config_data.get('learning_enabled', 'Not set')}")
        print(f"   - Allow Delegation: {config_data.get('allow_delegation', 'Not set')}")
        print(f"   - Verbose: {config_data.get('verbose', 'Not set')}")
        print(f"   - Max Iterations: {config_data.get('max_iter', 'Not set')}")
        print(f"   - Personality Traits: {config_data.get('personality_traits', 'Not set')}")
        print(f"   - Communication Style: {config_data.get('communication_formality', 'Not set')}")
        
        # Try to create the agent
        try:
            agent = dynamic_loader.create_agent_from_config(config_data, router)
            print(f"✅ Successfully created dynamic agent: {agent.role}")
            print(f"   - Agent verbose setting: {agent.verbose}")
            print(f"   - Agent allow_delegation: {agent.allow_delegation}")
            print(f"   - Agent max_iter: {agent.max_iter}")
            return True
        except Exception as e:
            print(f"❌ Failed to create agent: {e}")
            return False
    else:
        print(f"❌ No database config found for {test_role}")
        return False

def test_crew_integration():
    print("\n🚀 Testing Crew Integration...")
    
    try:
        from src.crews.internal.developer.crew import create_developer_crew
        from src.distributed_router import DistributedRouter
        
        router = DistributedRouter()
        inputs = {"project_name": "test", "working_dir": "/tmp"}
        
        print("Creating developer crew...")
        crew = create_developer_crew(router, inputs)
        
        print(f"✅ Developer crew created with {len(crew.agents)} agents:")
        for agent in crew.agents:
            print(f"   - {agent.role} (verbose: {agent.verbose}, delegation: {agent.allow_delegation})")
        
        return True
    except Exception as e:
        print(f"❌ Failed to create crew: {e}")
        return False

if __name__ == "__main__":
    print("=" * 60)
    print("🔍 ZeroAI Dynamic Agent Configuration Test")
    print("=" * 60)
    
    success1 = test_dynamic_agents()
    success2 = test_crew_integration()
    
    print("\n" + "=" * 60)
    if success1 and success2:
        print("✅ ALL TESTS PASSED - Agents are using database settings!")
    else:
        print("❌ SOME TESTS FAILED - Check configuration")
    print("=" * 60)

File: backup/www/admin/test_dynamic_agents.php
<?php 
$pageTitle = 'Test Dynamic Agents - ZeroAI';
$currentPage = 'agents';
include __DIR__ . '/includes/header.php';
require_once __DIR__ . '/../api/agent_db.php';

$agentDB = new AgentDB();

// Test dynamic agent loading
$testResults = [];

// Test 1: Check database agents
$agents = $agentDB->getAllAgents();
$testResults['database_agents'] = [
    'count' => count($agents),
    'agents' => array_slice($agents, 0, 3) // Show first 3 for brevity
];

// Test 2: Check if agents have all CrewAI options
$sampleAgent = !empty($agents) ? $agents[0] : null;
$testResults['crewai_options'] = [];
if ($sampleAgent) {
    $crewaiFields = [
        'memory', 'learning_enabled', 'learning_rate', 'personality_traits', 
        'communication_formality', 'allow_delegation', 'verbose', 'max_iter'
    ];
    foreach ($crewaiFields as $field) {
        $testResults['crewai_options'][$field] = isset($sampleAgent[$field]) ? 'Present' : 'Missing';
    }
}

// Test 3: Import agents to ensure database is populated
if (isset($_POST['import_agents'])) {
    $imported = $agentDB->importExistingAgents();
    $testResults['import_result'] = count($imported) . ' agents imported';
}

?>

<h1>🧪 Dynamic Agent Configuration Test</h1>

<div class="card">
    <h3>Test Results</h3>
    
    <h4>1. Database Agents</h4>
    <p><strong>Count:</strong> <?= $testResults['database_agents']['count'] ?></p>
    <?php if ($testResults['database_agents']['count'] > 0): ?>
        <p><strong>Sample Agents:</strong></p>
        <ul>
            <?php foreach ($testResults['database_agents']['agents'] as $agent): ?>
                <li><?= htmlspecialchars($agent['role']) ?> 
                    (Memory: <?= $agent['memory'] ? 'Yes' : 'No' ?>, 
                     Delegation: <?= $agent['allow_delegation'] ? 'Yes' : 'No' ?>,
                     Learning: <?= $agent['learning_enabled'] ? 'Yes' : 'No' ?>)
                </li>
            <?php endforeach; ?>
        </ul>
    <?php else: ?>
        <p style="color: orange;">⚠️ No agents found in database. Import agents first.</p>
    <?php endif; ?>
    
    <h4>2. CrewAI Options Check</h4>
    <?php if (!empty($testResults['crewai_options'])): ?>
        <table style="width: 100%; border-collapse: collapse;">
            <tr><th style="border: 1px solid #ddd; padding: 8px;">Field</th><th style="border: 1px solid #ddd; padding: 8px;">Status</th></tr>
            <?php foreach ($testResults['crewai_options'] as $field => $status): ?>
                <tr>
                    <td style="border: 1px solid #ddd; padding: 8px;"><?= htmlspecialchars($field) ?></td>
                    <td style="border: 1px solid #ddd; padding: 8px; color: <?= $status === 'Present' ? 'green' : 'red' ?>;">
                        <?= $status === 'Present' ? '✅' : '❌' ?> <?= $status ?>
                    </td>
                </tr>
            <?php endforeach; ?>
        </table>
    <?php else: ?>
        <p style="color: orange;">⚠️ No sample agent available for testing</p>
    <?php endif; ?>
    
    <?php if (isset($testResults['import_result'])): ?>
        <h4>3. Import Result</h4>
        <p style="color: green;">✅ <?= htmlspecialchars($testResults['import_result']) ?></p>
    <?php endif; ?>
</div>

<div class="card">
    <h3>Actions</h3>
    <form method="POST">
        <button type="submit" name="import_agents" class="btn-success">Import Agents from Static Files</button>
    </form>
    <br>
    <a href="/admin/agents" class="btn-primary">Go to Agent Management</a>
</div>

<div class="card">
    <h3>✅ Verification Status</h3>
    <?php 
    $hasAgents = $testResults['database_agents']['count'] > 0;
    $hasCrewAIOptions = !empty($testResults['crewai_options']) && 
                       in_array('Present', array_values($testResults['crewai_options']));
    ?>
    
    <div style="padding: 15px; border-radius: 5px; background: <?= $hasAgents && $hasCrewAIOptions ? '#d4edda' : '#fff3cd' ?>;">
        <?php if ($hasAgents && $hasCrewAIOptions): ?>
            <h4 style="color: #155724;">🎉 SUCCESS: Agents are using database settings!</h4>
            <p>✅ Database contains <?= $testResults['database_agents']['count'] ?> agents</p>
            <p>✅ Agents have all CrewAI configuration options</p>
            <p>✅ Dynamic loading is working correctly</p>
        <?php elseif ($hasAgents): ?>
            <h4 style="color: #856404;">⚠️ PARTIAL: Agents found but missing some options</h4>
            <p>✅ Database contains <?= $testResults['database_agents']['count'] ?> agents</p>
            <p>❌ Some CrewAI options may be missing</p>
        <?php else: ?>
            <h4 style="color: #721c24;">❌ FAILED: No agents in database</h4>
            <p>❌ Database is empty</p>
            <p>🔧 Click "Import Agents from Static Files" to populate database</p>
        <?php endif; ?>
    </div>
</div>

<div class="card">
    <h3>📋 How Dynamic Loading Works</h3>
    <ol>
        <li><strong>Database First:</strong> Crews try to load agents from database using <code>dynamic_agent_loader.py</code></li>
        <li><strong>Full Configuration:</strong> Database agents include all CrewAI options (memory, learning, personality, etc.)</li>
        <li><strong>Static Fallback:</strong> If database loading fails, crews fall back to static Python files</li>
        <li><strong>Admin Control:</strong> You can edit all agent settings through the admin interface</li>
    </ol>
    
    <h4>🔧 Files Updated for Dynamic Loading:</h4>
    <ul>
        <li><code>src/crews/internal/research/crew.py</code> - Research crew uses dynamic agents</li>
        <li><code>src/crews/internal/developer/crew.py</code> - Developer crew uses dynamic agents</li>
        <li><code>src/crews/internal/team_manager/agents.py</code> - Team manager loads dynamic coworkers</li>
        <li><code>src/utils/dynamic_agent_loader.py</code> - Handles all CrewAI options from database</li>
    </ul>
</div>

<?php include __DIR__ . '/includes/footer.php'; ?>